### PR TITLE
[AFR] Implemented Froghemoth

### PR DESCRIPTION
**Pull request by weirddan455**
Issue #7808 

I was a little unsure of how to handle the targeting of this ability.  We don't know ahead of time how many cards can be targeted or which player's graveyard.  I just did the targeting in checkTrigger but let me know if there is a better way.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/f/Froghemoth.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (super.checkTrigger(event, game)) {
+            Player controller = game.getPlayer(getControllerId());
+            Player damagedPlayer = game.getPlayer(event.getPlayerId());
+            if (controller != null && damagedPlayer != null) {
+                TargetCardInGraveyard target = new TargetCardInGraveyard(0, event.getAmount());
+                controller.chooseTarget(Outcome.Exile, damagedPlayer.getGraveyard(), target, this, game);
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/f/Froghemoth.java:84]: Do not use game dialogs or messages in `checkTrigger` or `applies` methods.

shot_210701_220411

**Comment by JayDi85** [comment]
I think you need `TargetAdjuster`. Search for `setTargetAdjuster` for examples.

**Comment by JayDi85** [comment]
Or just add new target for the triggered ability on check event (target card in xxx grave from 0 to yyy targets).

shot_210701_221417

**Comment by theelk801** [comment]
yeah just add the target in the `checkTrigger` method like in [[Blind Zealot]]

File: Mage.Sets/src/mage/cards/f/Froghemoth.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                    numCounters++;
+                } else {
+                    lifeGain++;
+                }
+            }
+        }
+        if (!cardsToExile.isEmpty()) {
+            controller.moveCards(cardsToExile, Zone.EXILED, source, game);
+            if (numCounters > 0) {
+                Permanent permanent = game.getPermanent(source.getSourceId());
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/f/Froghemoth.java:133]: this should use `source.getSourcePermanentIfItStillExists`

File: Mage.Sets/src/mage/cards/f/Froghemoth.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        }
+        return false;
+    }
+}
+
+class FroghemothEffect extends OneShotEffect {
+
+    public FroghemothEffect() {
+        super(Outcome.Exile);
+        this.staticText = "exile up to that many target cards from their graveyard. Put a +1/+1 counter on Froghemoth for each creature exiled this way. You gain 1 life for each noncreature card exiled this way";
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/f/Froghemoth.java:99]: don't forget to use `{this}` for the creature's name

**Review by theelk801 — CHANGES_REQUESTED** [review]

**Pull request merged**